common/requestfilter/rulesverifier_test.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package requestfilter_test
+
+import (
+	"errors"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/hyperledger/fabric-x-orderer/common/requestfilter"
+	"github.com/hyperledger/fabric-x-orderer/common/requestfilter/mocks"
+	"github.com/hyperledger/fabric-x-orderer/node/protos/comm"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRulesVerifier(t *testing.T) {
+	v := requestfilter.NewRulesVerifier(nil)
+	r := &mocks.FakeRule{}
+
+	t.Run("Verify Test", func(t *testing.T) {
+		r.VerifyReturns(nil)
+		v.AddRule(r)
+		err := v.Verify(&comm.Request{})
+		require.NoError(t, err)
+	})
+
+	t.Run("Verify Fail Test", func(t *testing.T) {
+		r.VerifyReturns(errors.New("verify error"))
+		v.AddRule(r)
+		err := v.Verify(&comm.Request{})
+		require.EqualError(t, err, "verify error")
+	})
+
+	t.Run("Update Test", func(t *testing.T) {
+		r.UpdateReturns(nil)
+		v.AddRule(r)
+		err := v.Update(&mocks.FakeFilterConfig{})
+		require.NoError(t, err)
+	})
+
+	t.Run("Verify Fail Test", func(t *testing.T) {
+		r.UpdateReturns(errors.New("update error"))
+		v.AddRule(r)
+		err := v.Update(&mocks.FakeFilterConfig{})
+		require.EqualError(t, err, "update error")
+	})
+}
+
+// scenario: test that verifiers can run in parallel.
+func TestConcurrentVeirfy(t *testing.T) {
+	v := requestfilter.NewRulesVerifier(nil)
+	fc := &mocks.FakeFilterConfig{}
+	fc.GetMaxSizeBytesReturns(1000, nil)
+	v.AddRule(requestfilter.NewMaxSizeFilter(fc))
+
+	var wg sync.WaitGroup
+	start := make(chan struct{})
+	end := make(chan struct{})
+	verifiers := 8
+
+	for i := 0; i < verifiers; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			<-start
+			err := v.Verify(&comm.Request{})
+			require.NoError(t, err)
+			time.Sleep(1 * time.Second)
+		}()
+	}
+
+	go func() {
+		close(start)
+		wg.Wait()
+		close(end)
+	}()
+
+	select {
+	case <-end:
+	case <-time.After(7 * time.Second):
+		t.Error("concurrent verify took too long")
+	}
```

Comment:
On normal machines you can make this assumption (all goroutines enter together, no more than 1s) but with loaded CI machines that has a (very small) potential to cause flakes. 

In addition, the code above this point does not really check the goroutine enter the rw lock together as the 1s delay is after you exit Verify. 
Another approach would be to use a mock filter that sleeps for a 1s when Verify is called.
The best approach would be to combine this with code that counts the maximal number of threads inside the Verify. Like:

enter verify, increment a (atomic) counter, 
sleep T
update (atomic) max-of-counter
sleep T 
decrement counter
exit verify

assert max-of-counter > 1

Having said all that, this is way too much... do it if you have some spare time in the next PR